packages/flutter_tools/test/analyze_once_test.dart

```diff
@@ -0,0 +1,196 @@
+      '// child: new Icon(Icons.add),',
+    );
+    source = source.replaceFirst(
+      '_counter++;',
+      '_counter++; throw "an error message";',
+    );
+    await libMain.writeAsString(source);
+
+    /// Analyze in the current directory - no arguments
+    fs.currentDirectory = tempDir;
+    await runCommand(
+      command: new AnalyzeCommand(),
+      arguments: <String>['analyze'],
+      statusTextContains: <String>[
+        'Analyzing',
+        '[warning] The parameter \'child\' is required',
+        '1 warning found.',
+      ],
+      // TODO(danrubel) fix dartanalyzer to have non-zero exit code
+      toolExit: false,
+    );
+  });
+
+  // Analyze a specific file outside the current directory
+  testUsingContext('flutter analyze one file with errors', () async {
+    fs.currentDirectory = originalDir;
+    await runCommand(
+      command: new AnalyzeCommand(),
+      arguments: <String>['analyze', libMain.path],
+      statusTextContains: <String>[
+        'Analyzing',
+        '[warning] The parameter \'child\' is required',
+        '1 warning found.',
+      ],
+      // TODO(danrubel) fix dartanalyzer to have non-zero exit code
+      toolExit: false,
+    );
+  });
+
+  // Analyze in the current directory - no arguments
+  testUsingContext('flutter analyze working directory with local options', () async {
+
+    // Insert an analysis_options.yaml file in the project
+    // which will trigger a lint for broken code that was inserted earlier
+    final File optionsFile = fs.file(fs.path.join(tempDir.path, 'analysis_options.yaml'));
+    await optionsFile.writeAsString('''
+include: package:flutter/analysis_options_user.yaml
+linter:
+  rules:
+    - only_throw_errors
+''');
+
+    /// Analyze in the current directory - no arguments
+    fs.currentDirectory = tempDir;
+    await runCommand(
+      command: new AnalyzeCommand(),
+      arguments: <String>['analyze'],
+      statusTextContains: <String>[
+        'Analyzing',
+        '[warning] The parameter \'child\' is required',
+        '[lint] Only throw instances of classes extending either Exception or Error',
+        '1 warning and 1 lint found.',
+      ],
+      // TODO(danrubel) fix dartanalyzer to have non-zero exit code
+      toolExit: false,
+    );
+  });
+
+  // Analyze a specific file outside the current directory
+  testUsingContext('flutter analyze one file with local options', () async {
+
+    /// Analyze a specific file outside the current directory
+    fs.currentDirectory = originalDir;
+    await runCommand(
+      command: new AnalyzeCommand(),
+      arguments: <String>['analyze', libMain.path],
+      statusTextContains: <String>[
+        'Analyzing',
+        '[warning] The parameter \'child\' is required',
+        '[lint] Only throw instances of classes extending either Exception or Error',
+        '1 warning and 1 lint found.',
+      ],
+      // TODO(danrubel) fix dartanalyzer to have non-zero exit code
```

Comment:
I updated this functionality to work around this issue so that `flutter analyze` will return a non-zero exit code even though `dartanalyzer` never does.